var/spack/repos/builtin/packages/py-isort/package.py

```diff
@@ -20,8 +21,10 @@ class PyIsort(PythonPackage):
 
     variant("colors", default=False, description="Install colorama for --color support")
 
-    depends_on("python@3.6.1:3", type=("build", "run"), when="@5:")
-    depends_on("python@2.7:2.8,3.4:", type=("build", "run"), when="@4.3:")
-    depends_on("python@2.6:2.8,3.3:", type=("build", "run"))
+    depends_on("python@3.8:", when="@5.12:", type=("build", "run"))
+    depends_on("python@3.6.1:3", when="@5:5.10", type=("build", "run"))
     depends_on("py-poetry-core@1:", type="build")
-    depends_on("py-colorama@0.4.3:0.4", type=("build", "run"), when="+colors")
+    depends_on("py-colorama@0.4.3:", when="+colors @5.12:", type=("build", "run"))
+    depends_on("py-colorama@0.4.3:0.4", when="+colors @:5.11", type=("build", "run"))
+
+    conflicts("^py-poetry-core@1.6.1:", when="@:5.11.3")
```

Comment:
```suggestion
    conflicts("^py-poetry-core@1.5:", when="@:5.11")
```
According to https://github.com/PyCQA/isort/issues/2077, the bug was introduced in poetry 1.5, not 1.6.1. And https://github.com/PyCQA/isort/pull/2078 was never backported to the 5.11 branch.

Reply:
I overlooked that they stated the version in the issue, thanks.
According to the Changelog (https://github.com/PyCQA/isort/blob/main/CHANGELOG.md#5115-january-30-2023-hotfix), they fixed it in 5.11.5 ( I did a typo in the 'when' here), but we can just restrict it to all of `5.11` to be on the safe side

Reply:
Oh nice, in that case let's just do `:5.11.4` since we know exactly when it was fixed